core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -168,6 +169,21 @@ private[scheduler] class BlacklistTracker (
       if (newTotal >= MAX_FAILURES_PER_EXEC && !executorIdToBlacklistStatus.contains(exec)) {
         logInfo(s"Blacklisting executor id: $exec because it has $newTotal" +
           s" task failures in successful task sets")
+        conf.get(config.BLACKLIST_ENABLED) match {
```

Comment:
Yes indeed. The only non-test usage of BlacklistTracker.isBlacklistEnabled is in the TaskSchedulerImpl's maybeCreateBlacklistTracker, which uses it as a condition for creating the BlacklistTracker at all. So I agree, we don't need a check here to see if the blacklist enabled, and if we did, isBlacklistEnabled would be a better choice.

However, it is still meaningful to see if spark.blacklist.kill is set. I believe this was just a typo: s/BLACKLIST_ENABLED/BLACKLIST_KILL_ENABLED/. This is fixed in my latest commit.